src/v/kafka/server/server.cc

```diff
@@ -699,29 +699,37 @@ ss::future<response_ptr> sasl_handshake_handler::handle(
     auto error = error_code::none;
 
     chunked_vector<ss::sstring> supported_sasl_mechanisms;
-    if (supports("SCRAM")) {
-        supported_sasl_mechanisms.emplace_back(
-          security::scram_sha256_authenticator::name);
-        supported_sasl_mechanisms.emplace_back(
-          security::scram_sha512_authenticator::name);
 
-        if (
-          request.data.mechanism
-          == security::scram_sha256_authenticator::name) {
-            ctx.sasl()->set_mechanism(
-              std::make_unique<security::scram_sha256_authenticator::auth>(
-                ctx.credentials()));
+    const auto add_scram_mechanisms =
+      [&supported_sasl_mechanisms, &request, &ctx]() {
+          supported_sasl_mechanisms.emplace_back(
+            security::scram_sha256_authenticator::name);
+          supported_sasl_mechanisms.emplace_back(
+            security::scram_sha512_authenticator::name);
+
+          if (
+            request.data.mechanism
+            == security::scram_sha256_authenticator::name) {
+              ctx.sasl()->set_mechanism(
+                std::make_unique<security::scram_sha256_authenticator::auth>(
+                  ctx.credentials()));
+
+          } else if (
+            request.data.mechanism
+            == security::scram_sha512_authenticator::name) {
+              ctx.sasl()->set_mechanism(
+                std::make_unique<security::scram_sha512_authenticator::auth>(
+                  ctx.credentials()));
+          }
+      };
 
-        } else if (
-          request.data.mechanism
-          == security::scram_sha512_authenticator::name) {
-            ctx.sasl()->set_mechanism(
-              std::make_unique<security::scram_sha512_authenticator::auth>(
-                ctx.credentials()));
-        }
+    if (supports("SCRAM")) {
+        add_scram_mechanisms();
     }
 
     if (supports("PLAIN")) {
```

Comment:
are these two mutually exclusive, or would there be need to deduplicate the resulting `supported_sasl_mechanisms` constainer?